src/ckeditor.jsx

```diff
@@ -5,24 +5,67 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
+import EditorWatchdog from '@ckeditor/ckeditor5-watchdog/src/editorwatchdog';
+import uid from '@ckeditor/ckeditor5-utils/src/uid';
+import { ContextWatchdogContext } from './context.jsx';
+import ContextWatchdog from '@ckeditor/ckeditor5-watchdog/src/contextwatchdog';
 
 export default class CKEditor extends React.Component {
 	constructor( props ) {
 		super( props );
 
 		// After mounting the editor, the variable will contain a reference to the created editor.
 		// @see: https://ckeditor.com/docs/ckeditor5/latest/api/module_core_editor_editor-Editor.html
-		this.editor = null;
 		this.domContainer = React.createRef();
+
+		/**
+		 * An instance of EditorWatchdog if the context watchdog was not passed via React's context. `null` otherwise.
+		 */
+		this.watchdog = null;
+
+		/**
+		 * An instance of ContextWatchdog passed from the React's context or `null` otherwise.
+		 */
+		this.contextWatchdog = null;
+
+		/**
+		 * A unique id for an editor component.
+		 */
+		this._id = uid();
+	}
+
+	get editor() {
+		if ( this.watchdog ) {
+			return this.watchdog.editor;
+		}
+
+		if ( this.contextWatchdog ) {
+			// TODO - try/catch should not be necessary as `getItem` could return `null` instead of throwing errors.
```

Comment:
I guess there is no ticket for this part of the code: https://github.com/ckeditor/ckeditor5/blob/3e6ea99fe28225c52092b621c3593748bb1c168e/packages/ckeditor5-watchdog/src/contextwatchdog.js#L418-L433

If you are going to change it in the future, would be good to attach a ticket id as a reference. Otherwise, I'd remove the TODO.